src/Paramore.Brighter.ServiceActivator/MessagePump.cs

```diff
@@ -132,17 +151,21 @@ public void Run()
                         Channel.Name, Thread.CurrentThread.ManagedThreadId);
 
                     RejectMessage(message);
+                    span?.SetStatus(ActivityStatusCode.Error,
+                        $"MessagePump: Stopping receiving of messages from {Channel.Name} on thread # {Thread.CurrentThread.ManagedThreadId}");
                     Channel.Dispose();
                     break;
                 }
                 catch (DeferMessageAction)
                 {
+                    span?.SetStatus(ActivityStatusCode.Error, "Deferring message for later action");
                     if (RequeueMessage(message)) continue;
                 }
                 catch (AggregateException aggregateException)
                 {
                     var (stop, requeue) = HandleProcessingException(aggregateException);
 
+                    span?.SetStatus(ActivityStatusCode.Error, $"Error while dispatching, Stopping {stop}, re-queueing {requeue}");
```

Comment:
Again, not sure what our action on stop vs requeue should be. We probably want a requeue, but unsure about stop. Do we just kill the span on a stop i.e. we add nothing because something else will re-process you?

Reply:
This won't necessarily "Kill" the whole span, the activity is created as a child of the activity in the producer. I will do a little more testing here to ensure.

At the minute I'm likening Service activator the asp.net as it's effectively a framework around requests, if something unexpected goes wrong in returns 500, if the upstream has a retry policy and the next attempt works then the parent will still return the operation as successful but when you look in the traces it'll have a failed attempt.

I'll setup a sample to fail and report back

Reply:
Thanks

Reply:
As discussed, this just shows as a failure of processing of that one part of the trace
![image](https://user-images.githubusercontent.com/7712876/193904065-34ae2479-4802-4baa-8907-127e9461e576.png)